datagen/procthor_datagen/datagen_runner_train.py

```diff
@@ -0,0 +1,1361 @@
+"""A script for generating rearrangement datasets."""
+
+import argparse
+import copy
+import json
+import math
+import multiprocessing as mp
+import os
+import platform
+import random
+import warnings
+from collections import defaultdict
+from typing import List, Set, Dict, Optional, Any, cast
+import time
+import sys
+
+import compress_pickle
+import numpy as np
+import torch.cuda
+import tqdm
+from allenact.utils.misc_utils import md5_hash_str_as_int
+from allenact_plugins.ithor_plugin.ithor_environment import IThorEnvironment
+
+from datagen.datagen_constants import OBJECT_TYPES_TO_NOT_MOVE
+from rearrange_constants import OPENNESS_THRESHOLD, IOU_THRESHOLD
+
+from datagen.procthor_datagen.datagen_utils import (
+    get_random_seeds,
+    filter_pickupable,
+    open_objs,
+    remove_objects_until_all_have_identical_meshes,
+    check_object_opens,
+    mapping_counts,
+)
+from utils.multiprocessing_utils import Manager, Worker, get_logger
+from rearrange.procthor_rearrange.constants import STARTER_DATA_DIR
+from rearrange.procthor_rearrange.environment import (
+    RearrangeProcTHOREnvironment,
+    RearrangeTaskSpec,
+)
+
+NUM_TRAIN_UNSEEN_EPISODES = 1_000  # 1 episode per scene
+NUM_TRAIN_SCENES = 10_000  # N episodes per scene
+NUM_VALID_SCENES = 1_000  # 1 episode per scene
+NUM_TEST_SCENES = 1_000  # 1 episode per scene
+
+SCENE_REUSE_COUNT = 20  # for training
+
+MAX_ROOMS_IN_HOUSE = 2
+
+MAX_POS_IN_HOUSE = 700
+
+MAX_TRIES = 40
+EXTENDED_TRIES = 10
+
+mp = mp.get_context("spawn")
+
+# Includes types used in both open and pickup actions:
+VALID_TARGET_TYPES = {
```

Comment:
Can we instead use `rearrange.constants.REARRANGE_SIM_OBJECTS`?

Reply:
I see this set is actually including `'Poster', 'RoomDecor', 'Window', 'Desktop', 'GarbageCan', 'Floor', 'BathtubBasin', 'PaperTowel', 'AppleSliced', 'Bed', 'PotatoSliced', 'Mirror', 'Dresser', 'Toaster', 'Desk', 'TargetCircle', 'StoveKnob', 'Sink', 'Television', 'ShowerHead', 'EggCracked', 'DiningTable', 'Shelf', 'SinkBasin', 'TVStand', 'Ottoman', 'ArmChair', 'LightSwitch', 'ShowerGlass', 'DeskLamp', 'VacuumCleaner', 'Curtains', 'CounterTop', 'StoveBurner', 'TowelHolder', 'Chair', 'Sofa', 'DogBed', 'BreadSliced', 'LettuceSliced', 'Painting', 'Bathtub', 'TomatoSliced', 'CoffeeTable', 'ToiletPaperHanger', 'GarbageBag', 'ShelvingUnit', 'Faucet', 'HandTowelHolder', 'HousePlant', 'SideTable', 'FloorLamp', 'Stool', 'CoffeeMachine'`, which are not potential targets given the action space, so I would not do that.